api/src/main/java/org/apache/iceberg/FileFormat.java

```diff
@@ -60,9 +60,10 @@ public String addExtension(String filename) {
   public static FileFormat fromFileName(CharSequence filename) {
     for (FileFormat format : VALUES) {
       int extStart = filename.length() - format.ext.length();
-      if (Comparators.charSequences()
-              .compare(format.ext, filename.subSequence(extStart, filename.length()))
-          == 0) {
+      if (extStart > 0
```

Comment:
Do we want to fail if the file name is just an extension? I think this is a reasonable choice because the only weird case I can imagine is setting your file name to metadata.json. All the other extensions are pretty much useless without another part to the name.

Reply:
Yeah, let's follow the common definition of an extension being "the bit after the last dot" (e.g., [commons-io](https://github.com/apache/commons-io/blob/4988e4b0697f6b6bdead3f3465c1e3c16fe2e33f/src/main/java/org/apache/commons/io/FilenameUtils.java#L629-L630)).

I think my test cases with `parquet` as filename were silly. Removed those

Reply:
I think if we did that we should just make a new version of the method, I don't think it's great to change the underlying perf semantics of the method in the API module. If it takes CharSequence we shouldn't change it to a string.